contracts/VoucherHub.sol

```diff
@@ -327,19 +327,37 @@ contract VoucherHub is
     }
 
     /**
-     * Get voucher address of a given account.
-     * Returns address(0) if voucher is not found.
-     * @param account voucher's owner address.
+     * Check if a voucher exists at a given address.
+     * @param account The address to be checked.
      */
-    function getVoucher(address account) external view returns (address voucherAddress) {
+    function isVoucher(address account) external view returns (bool) {
         VoucherHubStorage storage $ = _getVoucherHubStorage();
-        voucherAddress = Create2.computeAddress(
-            _getCreate2Salt(account), // salt
-            $._voucherCreationCodeHash // bytecode hash
-        );
+        return $._isVoucher[account];
+    }
+
+    /**
+     * Get the address of the voucher belonging to a given owner.
+     * Returns address(0) if voucher is not found.
+     * @param owner The owner of the voucher.
+     */
+    function getVoucher(address owner) external view returns (address voucherAddress) {
+        voucherAddress = predictVoucher(owner);
         return voucherAddress.code.length > 0 ? voucherAddress : address(0);
     }
 
+    /**
+     * Predict the address of the (created or not) voucher for a given owner.
+     * @param owner The owner of the voucher.
+     */
+    function predictVoucher(address owner) public view returns (address) {
```

Comment:
```suggestion
    function predictVoucherAddress(address owner) public view returns (address) {
```

Reply:
I dont think we want to update `getVoucher()`..
https://github.com/p00ls/contracts/blob/master/contracts/tokens/P00lsRegistryBase.sol#L60-L66

Reply:
Alright :+1: